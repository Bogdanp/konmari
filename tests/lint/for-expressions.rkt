#lang racket/base

(define foos
  (for/list ([i (in-range 10)]
             [i (in-range 20)])
    (define foo (* i 2))
    foo))

(define foo 42)
(define bar x)

(for/list ())
(for/list ((a '(1 2 3))) a)

(for/list ([x (in-range 10)] #:when y)
  (void))

(for/list ([x (in-range 10)] #:when y)
  x)

(for/lists (a b c)
           ([x (in-range 10)]
            [y (in-range 10)]
            [z (in-range 10)])
  (values
   (cons x a)
   (cons y b)
   (cons x a)))

(for/lists (a b c #:result c)
           ([x (in-range 10)]
            [y (in-range 10)]
            [z (in-range 10)])
  #:break z
  (values
   (cons x a)
   (cons y b)
   (cons x a)))

(for/fold ([xs null])
          ([x (in-list 10)])
  #t)

(for/fold ([xs null] #:result xs)
          ([x (in-list 10)])
  #t)

(for/fold ([xs null])
          ([x (in-list 10)])
  (cons x xs))
