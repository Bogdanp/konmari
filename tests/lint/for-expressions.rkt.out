/for-expressions.rkt:3:9:warning:identifier 'foos' is never used
/for-expressions.rkt:5:15:error:identifier 'i' is already defined in another clause
/for-expressions.rkt:9:9:warning:identifier 'foo' is never used
/for-expressions.rkt:10:9:warning:identifier 'bar' is never used
/for-expressions.rkt:12:1:error:for forms must contain at least one body expression
/for-expressions.rkt:13:12:error:'for' clauses should be surrounded by square brackets
/for-expressions.rkt:15:13:warning:identifier 'x' is never used
/for-expressions.rkt:21:17:warning:identifier 'c' is never used
/for-expressions.rkt:24:14:warning:identifier 'z' is never used
/for-expressions.rkt:40:13:warning:identifier 'xs' is never used
/for-expressions.rkt:41:13:warning:identifier 'x' is never used
/for-expressions.rkt:45:13:warning:identifier 'x' is never used
